lib/actions/local/configure.js

```diff
@@ -0,0 +1,285 @@
+	// to make sure that case defaults to `false`
+	data = _.assign(data, { persistentLogging: data.persistentLogging || false });
+
+	return inquirer
+		.prompt(inquirerOptions(data))
+		.then(answers => _.merge(data, answers));
+};
+
+// Taken from https://goo.gl/kr1kCt
+const CONNECTION_FILE = `\
+[connection]
+id=resin-wifi
+type=wifi
+
+[wifi]
+hidden=true
+mode=infrastructure
+ssid=My_Wifi_Ssid
+
+[wifi-security]
+auth-alg=open
+key-mgmt=wpa-psk
+psk=super_secret_wifi_password
+
+[ipv4]
+method=auto
+
+[ipv6]
+addr-gen-mode=stable-privacy
+method=auto\
+`;
+
+/*
+ * if the `resin-wifi` file exists (previously configured image or downloaded from the UI) it's used and reconfigured
+ * if the `resin-sample.ignore` exists it's copied to `resin-wifi`
+ * if the `resin-sample` exists it's reconfigured (legacy mode, will be removed eventually)
+ * otherwise, the new file is created
+ */
+const prepareConnectionFile = function(target) {
+	const _ = require('lodash');
+	const imagefs = require('resin-image-fs');
+
+	return imagefs
+		.listDirectory({
+			image: target,
+			partition: BOOT_PARTITION,
+			path: CONNECTIONS_FOLDER,
+		})
+		.then(function(files) {
+			// The required file already exists
+			if (_.includes(files, 'resin-wifi')) {
+				return null;
+			}
+
+			// Fresh image, new mode, accoding to https://github.com/balena-os/meta-balena/pull/770/files
+			if (_.includes(files, 'resin-sample.ignore')) {
+				return imagefs
+					.copy(
+						{
+							image: target,
+							partition: BOOT_PARTITION,
+							path: `${CONNECTIONS_FOLDER}/resin-sample.ignore`,
+						},
+						{
+							image: target,
+							partition: BOOT_PARTITION,
+							path: `${CONNECTIONS_FOLDER}/resin-wifi`,
+						},
+					)
+					.thenReturn(null);
+			}
+
+			// Legacy mode, to be removed later
+			// We return the file name override from this branch
+			// When it is removed the following cleanup should be done:
+			// * delete all the null returns from this method
+			// * turn `getConfigurationSchema` back into the constant, with the connection filename always being `resin-wifi`
+			// * drop the final `then` from this method
+			// * adapt the code in the main listener to not receive the config from this method, and use that constant instead
+			if (_.includes(files, 'resin-sample')) {
+				return 'resin-sample';
+			}
+
+			// In case there's no file at all (shouldn't happen normally, but the file might have been removed)
+			return imagefs
+				.writeFile(
+					{
+						image: target,
+						partition: BOOT_PARTITION,
+						path: `${CONNECTIONS_FOLDER}/resin-wifi`,
+					},
+					CONNECTION_FILE,
+				)
+				.thenReturn(null);
+		})
+		.then(connectionFileName => getConfigurationSchema(connectionFileName));
+};
+
+const removeHostname = function(schema) {
+	const _ = require('lodash');
+	return (schema.mapper = _.reject(schema.mapper, mapper =>
+		_.isEqual(Object.keys(mapper.template), ['hostname']),
+	));
+};
+
+export const configure = {
```

Comment:
I changed from `module.exports =` to `export const configure` as an artifact of switching to es module format